src/run/dependency/dependency-tree.ts

```diff
@@ -0,0 +1,92 @@
+import { simpleColouriser } from "../../helpers/colouriser.js";
+import DependencyNode from "../../model/DependencyNode.js";
+import AbstractDependencyGraph, { Logger, ServiceFinder } from "./AbstractDependencyGraph.js";
+
+export default class DependencyTree extends AbstractDependencyGraph {
```

Comment:
There are not tests for this, this is quite a complex component and would benefit from them.